pyomo/contrib/observer/model_observer.py

```diff
@@ -0,0 +1,1294 @@
+            (named_exprs, _vars, parameters, external_functions) = (
+                collect_components_from_expr(bnd)
+            )
+            if _vars:
+                raise NotImplementedError(
+                    'ModelChangeDetector does not support variables in the bounds of other variables'
+                )
+            if named_exprs:
+                raise NotImplementedError(
+                    'ModelChangeDetector does not support Expressions in the bounds of other variables'
+                )
+            if external_functions:
+                raise NotImplementedError(
+                    'ModelChangeDetector does not support external functions in the bounds of other variables'
+                )
+            ref_params.update(parameters)
+
+        _ref_params = self._params_referenced_by_var[v]
+        new_params = ref_params - _ref_params
+        old_params = _ref_params - ref_params
+
+        self._params_referenced_by_var[v] = ref_params
+
+        if new_params:
+            self._check_for_new_params(new_params)
+
+        for p in new_params:
+            self._referenced_params[p][3].add(v)
+        for p in old_params:
+            self._referenced_params[p][3].remove(v)
+
+        if old_params:
+            self._check_to_remove_params(old_params)
+
+    def _update_variables(self, variables: Optional[Collection[VarData]] = None):
+        if variables is None:
+            variables = self._vars
+        for v in variables:
+            _lb, _ub, _fixed, _domain_interval, _value = self._vars[v]
+            lb, ub, fixed, domain_interval, value = (
+                v._lb,
+                v._ub,
+                v.fixed,
+                v.domain.get_interval(),
+                v.value,
+            )
+            reason = Reason.no_change
+            if _fixed != fixed:
+                reason |= Reason.fixed
+            elif _fixed and (value != _value):
+                reason |= Reason.value
+            if lb is not _lb or ub is not _ub:
+                reason |= Reason.bounds
+            if _domain_interval != domain_interval:
+                reason |= Reason.domain
+            if reason:
+                self._updates.vars_to_update[v] |= reason
+                self._vars[v] = (lb, ub, fixed, domain_interval, value)
+                if reason & Reason.bounds:
+                    self._update_var_bounds(v)
+
+    def update_variables(self, variables: Optional[Collection[VarData]] = None):
+        self._update_variables(variables)
+        self._updates.run()
+
+    def _update_parameters(self, params: Optional[Collection[ParamData]] = None):
+        if params is None:
+            params = self._params
+        for p in params:
+            _val = self._params[p]
+            val = p.value
+            reason = Reason.no_change
+            if _val != val:
+                reason |= Reason.value
+            if reason:
+                self._updates.params_to_update[p] |= reason
```

Comment:
Cant this all be simplified to:
```suggestion
            if _val != val:
                self._updates.params_to_update[p] |= Reason.value
```